flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/OverTypeStrategy.java

```diff
@@ -182,18 +204,16 @@ private Optional<String> validateFollowingPreceding(
     public List<Signature> getExpectedSignatures(FunctionDefinition definition) {
         return Arrays.asList(
                 Signature.of(
```

Comment:
Does the signature still make sense?

Reply:
To be honest, this signature never made sense... :(

The entire `Over` expression is more of an internal concept that is hard to construct manually.

Reply:
Still... yes, it's a valid signature, just that the bounds can be of `NullType`.

Reply:
I would be fine to then just simplify the signature to something internal or return an empty list. It just spams the code.